internal/c/parts/compression/compression.cpp

```diff
@@ -7,52 +7,54 @@
 
 #include "compression.h"
 #include "qbs.h"
+
 #include "miniz.h"
 
+#include <vector>
+
 uint32_t func__adler32(qbs *text) {
-    if (!text->len) return 1;
-    return (uint32_t) adler32(1, text->chr, text->len);
+    if (!text->len)
+        return 1;
+    return (uint32_t)adler32(1, text->chr, text->len);
 }
 
 uint32_t func__crc32(qbs *text) {
-    if (!text->len) return 0;
-    return (uint32_t) crc32(0, text->chr, text->len);
+    if (!text->len)
+        return 0;
+    return (uint32_t)crc32(0, text->chr, text->len);
 }
 
 qbs *func__deflate(qbs *text) {
-    uLongf filesize = (uint32_t)text->len; // length of the text
-    uLongf compsize = compressBound(filesize);
-    unsigned char *dest = (unsigned char *)malloc(compsize);
-    int32_t result = compress(dest, &compsize, text->chr, filesize);
-    qbs *ret = qbs_new(compsize, 1);
-    memcpy(ret->chr, dest, compsize);
-    free(dest);
-    return ret;
+    auto fileSize = uLongf(text->len);
+    auto compSize = compressBound(fileSize);
+    auto dest = qbs_new(compSize, 1);                    // compressing directly to the qbs gives us a performance boost
+    compress(dest->chr, &compSize, text->chr, fileSize); // discard result because we do not do any error checking
+    qbs_left(dest, compSize);                            // discard the return value because it will return the same tmp qbs anyway
```

Comment:
Just write `return qbs_left(dest, compSize)`, you get to save a line and it avoids relying on `qbs_left` to perform a particular kind of optimisation

Reply:
Done.